### DISCUSSION TITLE
Using Prisma SQLite with Vercel

### DISCUSSION THREAD

**Discussion by pepyta** [vercel/vercel#4556] [Help] [answered]
Hello everyone!

I'm trying to store my user data on Vercel with SQLite (using Prisma), but I can't access other files inside `/pages/api` folder. How can I use the `/prisma/database.db` file inside an API endpoint?

```PrismaClientUnknownRequestError: 
Invalid `prisma.user.create()` invocation:
Error in connector: Error querying the database: unable to open database: /var/task/prisma/database.db
    at PrismaClientFetcher.request (/var/task/node_modules/@prisma/client/runtime/index.js:1:85980)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)```

**Comment by pepyta** [comment]
The project that I want to use the solution in is https://github.com/pepyta/lockify-web

**Comment by styfle** [comment, marked as answer]
Hi @pepyta 

The only writable directory on the filesystem is `/tmp`. Any temporary files you need to create in your Serverless Function must be in that directory (you can typically do `os.tmp()` to find the the temp directory on the current system.

Since sqlite stores data locally, we recommend using a different database that can share state between instances of your functions.

You can't reliably write to sqlite because storage is ephemeral in Serverless Functions. As your function receives more concurrent traffic, we will create new instances of your function that don't share the same storage.

**Reply by li-nd** [reply]
Hi.
How can I iclude my somesqlite.db to the file system of Serverless Function storage? I have a static database where the data does not change, but the api uses sql querys to search through it.

`["___vc_bridge.js","___vc_helpers.js","___vc_launcher.js","api","helpers","node_modules","package.json"]`
Tried, but it doesn't include db (/db/database.db) into the fs

**Reply by styfle** [reply]
You can try `const buffer = fs.readFileSync('../database.db')`

But I would suggest exporting your data [from sqlite to json](https://github.com/vercel/vercel/discussions/4443#discussioncomment-18785).

Then you won't need a separate dependency at runtime since Node.js can parse JSON natively.

**Reply by rblalock** [reply]
Interestingly enough ... I've been downloading a SQLIte file at build time in to a dir in the project and it's worked fine until I upgraded to NextJS 11.  I'm not sure quite why yet but I can consistently roll back to 10.x and it works, but when I deploy 11 to Vercel, it can't find that file anymore.
